demos/fleet_provisioning/fleet_provisioning_with_crt/fleet_provisioning_serializer.c

```diff
@@ -0,0 +1,469 @@
+                              const char * fmt,
+                              ... );
+/*-----------------------------------------------------------*/
+
+bool generateRegisterThingRequest( uint8_t * pBuffer,
+                                   size_t bufferLength,
+                                   const char * pCertificateOwnershipToken,
+                                   size_t certificateOwnershipTokenLength,
+                                   const char * pSerial,
+                                   size_t serialLength,
+                                   size_t * pOutLengthWritten )
+{
+    CborEncoder encoder, mapEncoder, parametersEncoder;
+    CborError cborRet;
+
+    assert( pBuffer != NULL );
+    assert( pCertificateOwnershipToken != NULL );
+    assert( pSerial != NULL );
+    assert( pOutLengthWritten != NULL );
+
+    /* For details on the RegisterThing request payload format, see:
+     * https://docs.aws.amazon.com/iot/latest/developerguide/fleet-provision-api.html#register-thing-request-payload
+     */
+    cbor_encoder_init( &encoder, pBuffer, bufferLength, 0 );
+    /* The RegisterThing request payload is a map with two keys. */
+    cborRet = cbor_encoder_create_map( &encoder, &mapEncoder, 2 );
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_stringz( &mapEncoder, "certificateOwnershipToken" );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_string( &mapEncoder, pCertificateOwnershipToken, certificateOwnershipTokenLength );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_stringz( &mapEncoder, "parameters" );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        /* Parameters in this example is length 1. */
+        cborRet = cbor_encoder_create_map( &mapEncoder, &parametersEncoder, 1 );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_stringz( &parametersEncoder, "SerialNumber" );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encode_text_string( &parametersEncoder, pSerial, serialLength );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encoder_close_container( &mapEncoder, &parametersEncoder );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        cborRet = cbor_encoder_close_container( &encoder, &mapEncoder );
+    }
+
+    if( cborRet == CborNoError )
+    {
+        *pOutLengthWritten = cbor_encoder_get_buffer_size( &encoder, ( uint8_t * ) pBuffer );
+    }
+    else
+    {
+        LogError( ( "Error during CBOR encoding: %s", cbor_error_string( cborRet ) ) );
+
+        if( ( cborRet & CborErrorOutOfMemory ) != 0 )
+        {
+            LogError( ( "Cannot fit RegisterThing request payload into buffer." ) );
+        }
+    }
+
+    return( cborRet == CborNoError );
+}
+
+/*-----------------------------------------------------------*/
+
+bool parseCrtResponse( const uint8_t * pResponse,
```

Comment:
Suggest creating new function in this file, not replace the parseCsrResponse. we can keep parseCsrResponse and parseCrtResponse together.